packages/next/next-server/server/config.ts

```diff
@@ -340,7 +340,26 @@ function assignDefaults(userConfig: { [key: string]: any }) {
         if (!item.defaultLocale) return true
         if (!item.domain || typeof item.domain !== 'string') return true
 
-        return false
+        let hasInvalidLocale = false
+
+        if (Array.isArray(item.locales)) {
+          for (const locale of item.locales) {
+            if (typeof locale !== 'string') hasInvalidLocale = true
+
+            for (const domainItem of i18n.domains) {
+              if (domainItem === item) continue
+              if (domainItem.locales && domainItem.locales.includes(locale)) {
+                console.warn(
+                  `Both ${item.domain} and ${domainItem.domain} configured the locale (${locale}) but only one can. Remove it from one i18n.domains config to continue`
+                )
+                hasInvalidLocale = true
+                break
```

Comment:
@Timer  What is we want to have 2 domains translated in the same language?
Exampe:
```
example.es 
  defaultLocale: 'es'
  locales: 'en'
example.fr 
  defaultLocale: 'fr'
  locales: 'en'
```

Is this possible?

Reply:
Another example more realistic. What if we have to domains for Belgium and Switzerland. In both countries `French` is a spoken language and we want to offer the content in french in `example.ch` and `example.be`.

Reply:
Hi, if you want the domain to be available for multiple domains you can **not** configure it for any domains and only in the main locales array and it will fallback to path based locale detection. Note: this can potentially negatively affect SEO since it can register as duplicate content on separate domains

Reply:
Hi @ijjk I opened an issue as recommended by @Timer :) https://github.com/vercel/next.js/issues/21892